packages/flutter/lib/src/material/debug.dart

```diff
@@ -4,66 +4,12 @@
 
 import 'package:flutter/widgets.dart';
 
-import 'material.dart';
 import 'material_localizations.dart';
 import 'scaffold.dart' show Scaffold, ScaffoldMessenger;
 
 // Examples can assume:
 // late BuildContext context;
 
-/// Asserts that the given context has a [Material] ancestor within the closest
-/// [LookupBoundary].
-///
-/// Used by many Material Design widgets to make sure that they are
-/// only used in contexts where they can print ink onto some material.
-///
-/// To call this function, use the following pattern, typically in the
-/// relevant Widget's build method:
-///
-/// ```dart
-/// assert(debugCheckHasMaterial(context));
-/// ```
-///
-/// Always place this before any early returns, so that the invariant is checked
-/// in all cases. This prevents bugs from hiding until a particular codepath is
-/// hit.
-///
-/// This method can be expensive (it walks the element tree).
-///
-/// Does nothing if asserts are disabled. Always returns true.
-bool debugCheckHasMaterial(BuildContext context) {
```

Comment:
Removing this will be a breaking change. It seems to be breaking many many Google tests.

Reply:
> I believe the only necessary breaking change is the MaterialInkController → InkController name change.

What a huge brain fart—you're absolutely right about `debugCheckHasMaterial`. I'll make sure this one is deprecated as well, thank you!